### PR TITLE
add initial Celery infrastructure

### PR DESCRIPTION
**Pull request by infinitewarp**
This branch adds the necessary _infrastructure_ to add Celery tasks and workers to the quipucords ecosystem, but it _doesn't_ add any _actual_ Celery task functions.

To confirm that tasks _could_ actually work with this branch, I locally defined two new "dummy" task functions as:

```diff
diff --git a/quipucords/scanner/task.py b/quipucords/scanner/task.py
index 7377ca7b..24036dec 100644
--- a/quipucords/scanner/task.py
+++ b/quipucords/scanner/task.py
@@ -4,6 +4,8 @@ from abc import ABCMeta, abstractmethod
 from multiprocessing import Value
 from typing import Tuple

+from celery import shared_task
+
 from api.models import ScanJob, ScanTask
 from scanner.exceptions import (
     ScanCancelException,
@@ -13,6 +15,18 @@ from scanner.exceptions import (
 )


+@shared_task(bind=True)
+def example_celery_task(self):
+    print(f"Request: {self.request!r}")
+
+
+@shared_task
+def example_scan_job_task(scan_job_id):
+    print(f"scan_job_id: {scan_job_id}")
+    scan_job = ScanJob.objects.get(id=scan_job_id)
+    print(f"ScanJob: {scan_job}")
+
+
 class ScanTaskRunner(metaclass=ABCMeta):
     """ScanTaskRunner is a logical breakdown of work."""
```

In one shell, I used the Django shell (`poetry run python quipucords/manage.py shell`) to spawn two instances:

```
Python 3.9.16 (main, Feb 22 2023, 13:06:47)
Type 'copyright', 'credits' or 'license' for more information
IPython 8.10.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from scanner.task import example_celery_task, example_scan_job_task

In [2]: example_celery_task.delay()
Out[2]: <AsyncResult: 9171f1a6-14ca-487e-8212-e2f373e7dea7>

In [3]: example_scan_job_task.delay(1)
Out[3]: <AsyncResult: dd34eae7-7fbf-45c1-8f99-f19b786232e5>
```

While in a second shell, I ran a Celery worker (`poetry run make celery-worker`) and observed the two task executions:

```
[2023-03-27 15:38:44,539: WARNING/ForkPoolWorker-8] Request: <Context: {'lang': 'py', 'task': 'scanner.task.example_celery_task', 'id': '9171f1a6-14ca-487e-8212-e2f373e7dea7', 'shadow': None, 'eta': None, 'expires': None, 'group': None, 'group_index': None, 'retries': 0, 'timelimit': [None, None], 'root_id': '9171f1a6-14ca-487e-8212-e2f373e7dea7', 'parent_id': None, 'argsrepr': '()', 'kwargsrepr': '{}', 'origin': 'gen79825@brasmith-shadowbook.lan', 'ignore_result': False, 'properties': {'correlation_id': '9171f1a6-14ca-487e-8212-e2f373e7dea7', 'reply_to': '9267085f-ac17-3d2c-a540-e8e980b6ef91', 'delivery_mode': 2, 'delivery_info': {'exchange': '', 'routing_key': 'celery'}, 'priority': 0, 'body_encoding': 'base64', 'delivery_tag': '3c95697c-7ce3-46e1-adfa-296acb86f927'}, 'reply_to': '9267085f-ac17-3d2c-a540-e8e980b6ef91', 'correlation_id': '9171f1a6-14ca-487e-8212-e2f373e7dea7', 'hostname': 'celery@brasmith-shadowbook.lan', 'delivery_info': {'exchange': '', 'routing_key': 'celery', 'priority': 0, 'redelivered': None}, 'args': [], 'kwargs': {}, 'is_eager': False, 'callbacks': None, 'errbacks': None, 'chain': None, 'chord': None, 'called_directly': False, '_protected': 1}>
[2023-03-27 15:38:54,245: WARNING/ForkPoolWorker-8] scan_job_id: 1
[2023-03-27 15:38:54,261: WARNING/ForkPoolWorker-8] ScanJob: {id:1, scan:None, scan_type:connect, status:completed, options: None, report_id: None, start_time: 2023-02-21 19:27:26.235180, end_time: 2023-02-21 19:27:28.256759, connection_results: 1, inspection_results: None}
```

### CODE REVIEW

File: requirements-build.txt [hunk 1]
```diff
@@ -10,14 +10,23 @@ cython==0.29.33
     # via -r requirements-build.in
 docutils==0.19
     # via -r requirements-build.in
+editables==0.3
+    # via hatchling
 flit-core==3.8.0
     # via -r requirements-build.in
+hatchling==1.13.0
+    # via -r requirements-build.in
```
Comment (bruno-fs) [inline requirements-build.txt:18]: oh boy. hatchling depends on itself for building.

looking forward to see how this bad boy will behave downstream (hopefully no changes required)

Comment (bruno-fs) [reply requirements-build.txt:18]: (not that this is any blocker. just saying)

Comment (bruno-fs) [reply requirements-build.txt:18]: OMG docs don't have info on how to build it lol
image

File: Makefile [hunk 2]
```diff
@@ -129,6 +130,9 @@ server-makemigrations:
 server-migrate:
 	$(PYTHON) quipucords/manage.py migrate --settings quipucords.settings -v 3
 
+celery-worker:
```
[the remark this answers is not in the document]
Comment (bruno-fs) [reply Makefile:133]: :ok_person: OK, makes sense.

File: requirements-build.txt (hunk 1 above)
Comment (infinitewarp) [reply requirements-build.txt:18]: I guess we'll find out downstream soon enough! 🚀 💥

**Pull request merged**